### ISSUE TITLE
ERROR: Primitive '_FileCopy' failed. when starting scide - boost needs upgrade to higher version to fix

### ISSUE THREAD

**Issue by shimpe** [supercollider/supercollider#5865]
My /home is mounted on a different physical disk than /usr. 

When starting scide, I get the following error:
```
*** Welcome to SuperCollider 3.13.0-dev. *** For help press Ctrl-D.
Couldn't set realtime scheduling priority 1: Operation not permitted
SCDoc: Indexing help-files...
ERROR: Primitive '_FileCopy' failed.
caught exception 'Invalid cross-device link: copy from "/usr/local/share/SuperCollider/HelpSource/browse.css" to "/home/shimpe/.local/share/SuperCollider/Help/browse.css"' in primitive in method Meta_File:copy
RECEIVER:
class File (0x563d1fd95400) {
  instance variables [19]
    name : Symbol 'File'
    nextclass : instance of Meta_FileDialog (0x563d1f9419c0, size=19, set=5)
    superclass : Symbol 'UnixFILE'
    subclasses : nil
    methods : instance of Array (0x563d1fd95580, size=12, set=4)
    instVarNames : instance of SymbolArray (0x563d1fd95700, size=1, set=2)
    classVarNames : instance of SymbolArray (0x563d1fd95880, size=1, set=2)
    iprototype : instance of Array (0x563d1fd957c0, size=1, set=2)
    cprototype : instance of Array (0x563d1fd95940, size=1, set=2)
    constNames : nil
    constValues : nil
    instanceFormat : Integer 0
    instanceFlags : Integer 0
    classIndex : Integer 1077
    classFlags : Integer 0
    maxSubclassIndex : Integer 1077
    filenameSymbol : Symbol '/usr/local/share/SuperCollider/SCClassLibrary/Common/Files/File.sc'
    charPos : Integer 0
    classVarIndex : Integer 212
}

PROTECTED CALL STACK:
	Meta_MethodError:new	0x563d1f733700
		arg this = PrimitiveFailedError
		arg what = caught exception 'Invalid cross-device link: copy from "/usr/local/share/SuperCollider/HelpSource/browse.css" to "/home/shimpe/.local/share/SuperCollider/Help/browse.css"' in primitive in method Meta_File:copy
		arg receiver = File
	Meta_PrimitiveFailedError:new	0x563d1f73b540
		arg this = PrimitiveFailedError
		arg receiver = File
	Object:primitiveFailed	0x563d1f158d00
		arg this = File
	ArrayedCollection:do	0x563d20200ac0
		arg this = [ [ /usr/local/share/SuperCollider/HelpSource/BrokenLink.html, BrokenLink.html ], [ /usr/local/share/SuperCollider/HelpSource/browse.css, browse.css ], [ /usr/local/share/SuperCollider/HelpSource/Browse.html, Browse.html ], [ /usr/local/share/SuperCollider/HelpSource/browse.js, browse.js ], [ /usr/local/share/SuperCollider/HelpSource/codemirror.css, codemirror.css ], [ /usr/local/share/SuperCollider/HelpSource/editor.css, editor.css ], [ /usr/local/share/SuperCollider/HelpSource/editor.js, editor.js ], [ /u...etc...
		arg function = a Function
		var i = 1
	Meta_SCDoc:indexAllDocuments	0x563d1f65afc0
		arg this = SCDoc
		arg clearCache = false
		var now = 0.549377424
		var key = Classes/String
		var doc = SCDocEntry("Classes/QKeyModifiers", "QKeyModifiers", "(Undocumented class)")
		var nonHelpFiles = [ [ /usr/local/share/SuperCollider/HelpSource/BrokenLink.html, BrokenLink.html ], [ /usr/local/share/SuperCollider/HelpSource/browse.css, browse.css ], [ /usr/local/share/SuperCollider/HelpSource/Browse.html, Browse.html ], [ /usr/local/share/SuperCollider/HelpSource/browse.js, browse.js ], [ /usr/local/share/SuperCollider/HelpSource/codemirror.css, codemirror.css ], [ /usr/local/share/SuperCollider/HelpSource/editor.css, editor.css ], [ /usr/local/share/SuperCollider/HelpSource/editor.js, editor.js ], [ /u...etc...
		var undocClasses = IdentitySet[ BelaServerOptions, Range, PureMultiOutUGen, QDragSink, BinaryOpXStream, NodeID, DrawGridX, AbstractGroup, MIDIEvent, MethodQuote, QCurve, OutOfContextReturnError, QTextViewBase, QDialog, ScopeBuffer, FuncStreamAsRoutine, Penv, PatternControl, HIDFuncDeviceProtoMatcher, QCallback, QFont, QRenderProcessTerminationStatus, PlusFreqScopeWindow, HIDElementProtoDispatcher, MIDISMPTEAssembler, QScope, QScope2, CurveWarp, QColorGroup, BlankGridLines, QNumberBox, CleanupStream, TextArchiveMDPlugin, HIDRa...etc...
		var additions = Dictionary[ (Classes/String -> [ /usr/local/share/SuperCollider/HelpSource/Classes/String.ext.schelp ]) ]
	Meta_SCDoc:documents	0x563d1f65e980
		arg this = SCDoc
	Meta_SCDoc:prepareHelpForURL	0x563d1f662c00
		arg this = SCDoc
		arg url = file:///home/shimpe/.local/share/SuperCollider/Help/Help.html
		var path = /home/shimpe/.local/share/SuperCollider/Help/Help.html
		var targetBasePath = /home/shimpe/.local/share/SuperCollider/Help
		var pathIsCaseInsensitive = false
		var subtarget = Help
		var src = nil
		var c = nil
		var cmd = nil
		var doc = nil
		var destExist = false
		var destMtime = nil
		var verpath = /home/shimpe/.local/share/SuperCollider/Help/version
	a FunctionDef	0x563d1efa4400
		sourceCode = "<an open Function>"
	Function:prTry	0x563d1f9c1100
		arg this = a Function
		var result = nil
		var thread = a Routine
		var next = nil
		var wasInProtectedFunc = true
	
CALL STACK:
	MethodError:reportError
		arg this = <instance of PrimitiveFailedError>
	< closed FunctionDef >
		arg error = <instance of PrimitiveFailedError>
	Integer:forBy
		arg this = 0
		arg endval = 0
		arg stepval = 2
		arg function = <instance of Function>
		var i = 0
		var j = 0
	SequenceableCollection:pairsDo
		arg this = [*2]
		arg function = <instance of Function>
	Scheduler:seconds_
		arg this = <instance of Scheduler>
		arg newSeconds = 0.742669093
	Meta_AppClock:tick
		arg this = <instance of Meta_AppClock>
		var saveClock = <instance of Meta_SystemClock>
	Process:tick
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Primitive '_FileCopy' failed.
caught exception 'Invalid cross-device link: copy from "/usr/local/share/SuperCollider/HelpSource/browse.css" to "/home/shimpe/.local/share/SuperCollider/Help/browse.css"' in primitive in method Meta_File:copy
RECEIVER: File
```

**Comment by shimpe** [comment]
I just cloned and built latest develop (5893b7c387e6d2affe83617311423f75a651ada0)

**Comment by shimpe** [comment]
I should have mentioned: this is on a linux computer. Doing a "manual" copy (via bash command line) for this file makes this specific error disappear (but then it complains about the next copy operation that fails)https://github.com/supercollider/sc3-plugins

**Comment by shimpe** [comment]
Ah... probably it's this bug: https://github.com/boostorg/filesystem/issues/254
Solved in boost 1.77.0 and later. I see supercollider is using 1.74.0.

**Comment by pierrepierrepierre** [comment]
Hi, I'm running trhough same issue after upgrading linux mint 21...
Any idea on how to make the doc usable inside the app?
thanks

**Comment by shimpe** [comment]
@pierrepierrepierre if you don't mind building your own supercollider from code, you can try with https://github.com/shimpe/supercollider/tree/topic/upgrade_boost_1.80.0_unpatched 
it-works-for-me :)

**Comment by garpu** [comment]
https://github.com/supercollider/supercollider/issues/5982 I bumped into this one this morning.  Alien Bob (who maintains the Slackware supercollider package) had it not building against the system boost (currently for slackware-current at 1.81)  To test, I deleted Help in .local/share/SuperCollider, and it recreated the Help directory properly when compiled against my system boost.

**Comment by pierrepierrepierre** [comment]
thanks @shimpe  & @garpu ,
I found some sort of workaround by changing the runtime directory to : /usr/share/SuperCollider
and it works fine now...

**Comment by FabienCesari** [comment]
I also encountered that issue because my `/home` and `/` are on different partitions. I went around it by changing the install location in cmake. 

`cmake -DCMAKE_BUILD_TYPE=Release -DNATIVE=ON -DCMAKE_INSTALL_PREFIX=~/.local ..`

**Comment by PechenkaJun** [comment]
I had the same issue on SuperCollider 3.11.2. I solved it by changing access in /usr/share/SuperCollider/

`cd /usr/share/SuperCollider/`

`sudo chmod -R 777 SuperCollider/`

**Comment by TheOtherDoctor** [comment]
> I had the same issue on SuperCollider 3.11.2. I solved it by changing access in /usr/share/SuperCollider/
> 
> `cd /usr/share/SuperCollider/`
> 
> `sudo chmod -R 777 SuperCollider/`

Having the same issue here, just wondering - SC IDE needs _write_ access to the /usr/share/SuperCollider?

And I guess your post has a typo, since there is no /usr/share/SuperCollider/SuperCollider/ - thus I tried the chmod for ./HelpSource instead, which did _not_ give any improvements for me., neither did doing the 777 for whole /usr/share/SuperCollider. 

Without having deep understanding what SC does to the help files in the start phase, I gave it a try to (re)move the HOME/.local/share/SuperCollider dirtree, but that also fails. I am not so motivated to compile SC from scratch, as I expect missing up things further.
 
And yes the help system in the IDE is defunct still, making usage mostly impossible for me as a newbie in SC.
Any help would be deeply appreciated. 

Currently failing with 3.11.2 on Ubuntu 22.04 LTS.

**Comment by FabienCesari** [comment]
> 

FWIW the [wiki](https://github.com/supercollider/supercollider/wiki/Installing-SuperCollider-from-source-on-Ubuntu) is really well done if you'd wanted to try building it from scratch. It doesn't take that long

**Comment by dyfer** [comment]
Thanks for linking the wiki page @FabienCesari 
For building SC, the [platform readme](https://github.com/supercollider/supercollider/blob/develop/README_LINUX.md) should be the primary document. Is the wiki page you linked easier to follow or in other ways better than that document? If so, please let us know, it would be great to update the linux readme if needed.

**Comment by FabienCesari** [comment]
@dyfer good to know, thanks! I haven't tried the readme, I usually go to the wiki directly, I find it pretty handy.

**Comment by yeeking** [comment]
Hello!

My solution is this:
* run scide
* in bash
```
# delete the faulty folder just created by scide
rm -rf ~/.local/share/SuperCollider/Help 
# manually recreate
cp -rL /usr/share/SuperCollider/HelpSource  ~/.local/share/SuperCollider/Help 
```
* in scide, reboot the interpreter.

**Comment by tank-trax** [comment]
>
> ```
> # delete the faulty folder just created by scide
> rm -rf ~/.local/share/SuperCollider/Help 
> # manually recreate
> cp -rL /usr/share/SuperCollider/HelpSource  ~/.local/share/SuperCollider/Help 
>```


thank you, this worked for me on Debian 12
SuperCollider 3.13.0

**Comment by hr30** [comment]
I had the same problem with SuperCollider 3.13.0, but here it happened with the extensions. Then copied copied manually as proposed by tank-trax:

```
# delete the faulty folder just created by scide
rm -rf ~/.local/share/SuperCollider/Extensions
# manually recreate
cp -rL /usr/share/SuperCollider/HelpSource  ~/.local/share/SuperCollider/Extensions
```

When starting the IDE, I got a lot of error messages saying, that UGens are duplicated and finally the library wasn't compiled successfully.
The main issue here was, that I also installed the SC3 plug-ins  3.9.1 from the Synaptic program. When uninstalling the SC3-plugin packages, SC compiles the library successfully. 
Are the SC3-plugins available? They are and seem integrated  SC 3.13.0 now - at least in the deb-package of SuperCollider for Ubuntu Studio 23.10.

Does this help too?
